File: library/src/main/java/com/flipkart/okhttpstats/toolbox/PreferenceManager.java
/*
 *
 *  * Copyright 2016 Flipkart Internet Pvt. Ltd.
 *  *
 *  * Licensed under the Apache License, Version 2.0 (the "License");
 *  * you may not use this file except in compliance with the License.
 *  * You may obtain a copy of the License at
 *  *
 *  *    http://www.apache.org/licenses/LICENSE-2.0
 *  *
 *  * Unless required by applicable law or agreed to in writing, software
 *  * distributed under the License is distributed on an "AS IS" BASIS,
 *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  * See the License for the specific language governing permissions and
 *  * limitations under the License.
 *
 */

package com.flipkart.okhttpstats.toolbox;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Preference Manager to store network speed
 */
public class PreferenceManager {

    private static final String PREFERENCES = "PREFERENCES";
    private final SharedPreferences mSharedPreferences;

    public PreferenceManager(Context context) {
        this.mSharedPreferences = context.getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE);
    }

    public void setAverageSpeed(String networkType, float avgSpeed) {
        SharedPreferences.Editor editor = mSharedPreferences.edit();
        editor.putFloat(networkType, avgSpeed);
        editor.apply();
    }

    public float getAverageSpeed(String networkType) {
        return this.mSharedPreferences.getFloat(networkType, 0F);
    }
}

File: library/src/main/java/com/flipkart/okhttpstats/toolbox/PreferenceManager.kt
package com.flipkart.okhttpstats.toolbox

/**
 * Preference Manager to store network speed
 */
class PreferenceManager(context: android.content.Context) {
    private val sharedPreferences: android.content.SharedPreferences

    init {
        this.sharedPreferences = context.getSharedPreferences(PreferenceManager.Companion.PREFERENCES, android.content.Context.MODE_PRIVATE)
    }

    fun setAverageSpeed(networkType: String, avgSpeed: Float) {
        val editor = sharedPreferences.edit()
        editor.putFloat(networkType, avgSpeed)
        editor.apply()
    }

    fun getAverageSpeed(networkType: String): Float {
        return this.sharedPreferences.getFloat(networkType, 0f)
    }

    companion object {
        private val PREFERENCES = "PREFERENCES"
    }
}